### PR TITLE
Merge 1.7.7.x into develop - 06/05/2020

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Merge 1.7.7.x in develop
| Type?         | improvement
| Category?     | ME
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | nothing to test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18998)

### CODE REVIEW

**Comment by matks** [comment]
@boubkerbribri I add git conflicts on the following files:
```
	both modified:   tests/puppeteer/pages/BO/design/pages/index.js
	both modified:   tests/puppeteer/pages/BO/international/currencies/index.js
	both modified:   tests/puppeteer/pages/BO/international/languages/index.js
	both modified:   tests/puppeteer/pages/BO/international/taxes/index.js
	both modified:   tests/puppeteer/pages/BO/shopParameters/contact/index.js
	both modified:   tests/puppeteer/pages/BO/shopParameters/trafficAndSeo/seoAndUrls/index.js
```

Please check to tell me if I solved the conflicts in a good way.

File: tests/puppeteer/pages/BO/shopParameters/trafficAndSeo/seoAndUrls/index.js [hunk 1]
```diff
@@ -79,17 +75,12 @@ module.exports = class SeoAndUrls extends BOBasePage {
...
+        `${this.dropdownToggleButton(row)}[aria-expanded='true']`,
       ),
     ]);
-    // Click on delete and wait for modal
-    await Promise.all([
-      this.page.click(this.deleteRowLink.replace('%ROW', row)),
-      this.waitForVisibleSelector(`${this.confirmDeleteModal}.show`),
-    ]);
-    await this.confirmDeleteSeoUrlPage();
+    await this.clickAndWaitForNavigation(this.deleteRowLink(row));
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/shopParameters/trafficAndSeo/seoAndUrls/index.js]:
```suggestion
    // Click on delete and wait for modal
    await Promise.all([
      this.page.click(this.deleteRowLink(row)),
      this.waitForVisibleSelector(`${this.confirmDeleteModal}.show`),
    ]);
    await this.confirmDeleteSeoUrlPage();
```

File: tests/puppeteer/pages/BO/shopParameters/contact/index.js [hunk 2]
```diff
@@ -135,17 +131,14 @@ module.exports = class Contacts extends BOBasePage {
...
       ),
     ]);
     // Click on delete
-    await Promise.all([
-      this.page.click(this.deleteRowLink.replace('%ROW', row)),
-      this.waitForVisibleSelector(`${this.confirmDeleteModal}.show`),
-    ]);
-    await this.confirmDeleteContact();
+
+    await this.clickAndWaitForNavigation(this.deleteRowLink(row));
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/shopParameters/contact/index.js]:
```suggestion
    await Promise.all([
      this.page.click(this.deleteRowLink(row)),
      this.waitForVisibleSelector(`${this.confirmDeleteModal}.show`),
    ]);
    await this.confirmDeleteContact();
```

File: tests/puppeteer/pages/BO/international/taxes/index.js [hunk 3]
```diff
@@ -195,19 +184,12 @@ module.exports = class Taxes extends BOBasePage {
...
     ]);
-    // Click on delete and wait for modal
-    await Promise.all([
-      this.page.click(this.taxesGridDeleteLink.replace('%ROW', row).replace('%COLUMN', 'actions')),
-      this.waitForVisibleSelector(`${this.confirmDeleteModal}.show`),
-    ]);
-    await this.confirmDeleteTaxes();
+
+    // Click on delete
+    await this.clickAndWaitForNavigation(this.taxesGridDeleteLink(row));
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/international/taxes/index.js]:
```suggestion
    // Click on delete and wait for modal
    await Promise.all([
      this.page.click(this.taxesGridDeleteLink(row)),
      this.waitForVisibleSelector(`${this.confirmDeleteModal}.show`),
    ]);
    await this.confirmDeleteTaxes();
```

File: tests/puppeteer/pages/BO/international/languages/index.js [hunk 4]
```diff
@@ -149,17 +145,13 @@ module.exports = class Languages extends LocalizationBasePage {
...
       ),
     ]);
-    // Click on delete and wait for modal
-    await Promise.all([
-      this.page.click(this.deleteRowLink.replace('%ROW', row)),
-      this.waitForVisibleSelector(`${this.confirmDeleteModal}.show`),
-    ]);
-    await this.confirmDeleteLanguages(this.bulkActionsDeleteButton);
+
+    await this.clickAndWaitForNavigation(this.deleteRowLink(row));
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/international/languages/index.js]:
```suggestion
    // Click on delete and wait for modal
    await Promise.all([
      this.page.click(this.deleteRowLink(row)),
      this.waitForVisibleSelector(`${this.confirmDeleteModal}.show`),
    ]);
    await this.confirmDeleteLanguages(this.bulkActionsDeleteButton);
```

File: tests/puppeteer/pages/BO/international/currencies/index.js [hunk 5]
```diff
@@ -177,17 +166,13 @@ module.exports = class Currencies extends LocalizationBasePage {
...
       ),
     ]);
-    // Click on delete and wait for modal
-    await Promise.all([
-      this.page.click(this.deleteRowLink.replace('%ROW', row)),
-      this.waitForVisibleSelector(`${this.confirmDeleteModal}.show`),
-    ]);
-    await this.confirmDeleteCurrency();
+
+    await this.clickAndWaitForNavigation(this.deleteRowLink(row));
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/international/currencies/index.js]:
```suggestion
    // Click on delete and wait for modal
    await Promise.all([
      this.page.click(this.deleteRowLink(row)),
      this.waitForVisibleSelector(`${this.confirmDeleteModal}.show`),
    ]);
    await this.confirmDeleteCurrency();
```

File: tests/puppeteer/pages/BO/design/pages/index.js [hunk 6]
```diff
@@ -108,15 +116,12 @@ module.exports = class Pages extends BOBasePage {
...
+      this.waitForVisibleSelector(`${this.listTableToggleDropDown(table, row)}[aria-expanded='true']`),
     ]);
     // Click on delete and wait for modal
-    await Promise.all([
-      this.page.click(deleteRowLink.replace('%ROW', row)),
-      this.waitForVisibleSelector(`${confirmDeleteModal}.show`),
-    ]);
-    await this.confirmDeleteFromTable(table);
+    this.dialogListener();
+    await this.clickAndWaitForNavigation(this.deleteRowLink(table, row));
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/design/pages/index.js]:
```suggestion
    await Promise.all([
      this.page.click(this.deleteRowLink(row)),
      this.waitForVisibleSelector(`${this.confirmDeleteModal(table)}.show`),
    ]);
    await this.confirmDeleteFromTable(table);
```

File: tests/puppeteer/pages/BO/design/pages/index.js [hunk 7]
```diff
@@ -156,8 +157,8 @@ module.exports = class Pages extends BOBasePage {
    * @param table
    * @return {Promise<void>}
    */
-  async confirmDeleteFromTable(table) {
-    await this.clickAndWaitForNavigation(this.confirmDeleteButton.replace('%TABLE', table));
+  async confirmDeleteWithBulkActions(table) {
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/design/pages/index.js]:
```suggestion
  async confirmDeleteFromTable(table) {
```

**Review by matthieu-rolland — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @matks

**Pull request merged**